pennylane_lightning/src/util/Util.hpp

```diff
@@ -214,17 +212,17 @@ std::complex<T> innerProd(const std::complex<T> *data_1,
                           const size_t data_size) {
     std::complex<T> result(0, 0);
 
-    if constexpr (USE_CBLAS) {
-        if constexpr (std::is_same_v<T, float>)
-            result = cblas_cdotu_sub(data_size, data_1, 1, data_2, 1, &result);
-        else if constexpr (std::is_same_v<T, double>)
-            result = cblas_zdotu_sub(data_size, data_1, 1, data_2, 1, &result);
-    } else {
-        result = std::inner_product(
-            data_1, data_1 + data_size, data_2, std::complex<T>(), ConstSum<T>,
-            static_cast<std::complex<T> (*)(std::complex<T>, std::complex<T>)>(
-                &ConstMult<T>));
-    }
+#ifdef USE_CBLAS
```

Comment:
Is there a good reason to favour the preprocessor rather than the `if constexpr` support for this?

Reply:
The generated assemblies from using `if constexpr` and the preprocessor are identical! I updated this part mainly to emphasise the use of USE_CBLAS within the code, and I thought that making use of preprocessors are more common for such situations. I can simply revert it back to what it was if you think differently?

Reply:
Sounds good. I'd prefer to avoid use of the preprocessor unless we need to use it (or unless it makes things much simpler).  In this case given that everything happens at build time anyway, I'd rather we keep to C++ constructs where we can.